tuftool/src/rhtas.rs

```diff
@@ -0,0 +1,420 @@
+                target
+                    .custom
+                    .insert("sigstore".to_string(), custom_sigstore_metadata);
+                editor
+                    .add_target(target_name.clone(), target.clone())
+                    .context(error::DelegationStructureSnafu)?;
+            }
+        }
+        Ok(())
+    }
+
+    async fn set_ctlog_target(&self, editor: &mut RepositoryEditor) -> Result<()> {
+        if let Some(ref ctlog_target_path) = self.ctlog_target {
+            let mut ctlog_target = build_targets(ctlog_target_path, self.follow).await?;
+
+            if self.ctlog_status != "Active" && self.ctlog_status != "Expired" {
+                return error::NoValidTargetStatusSnafu {}.fail();
+            }
+            let custom_sigstore_metadata = json!({
+                "status": self.ctlog_status,
+                "uri": self.ctlog_uri,
+                "usage": "CTFE"
+            });
+
+            if let Some((target_name, target)) = ctlog_target.iter_mut().next() {
+                target
+                    .custom
+                    .insert("sigstore".to_string(), custom_sigstore_metadata);
+                editor
+                    .add_target(target_name.clone(), target.clone())
+                    .context(error::DelegationStructureSnafu)?;
+            }
+        }
+        Ok(())
+    }
+
+    async fn set_rekor_target(&self, editor: &mut RepositoryEditor) -> Result<()> {
+        if let Some(ref rekor_target_path) = self.rekor_target {
+            let mut rekor_target = build_targets(rekor_target_path, self.follow).await?;
+
+            if self.rekor_status != "Active" && self.rekor_status != "Expired" {
+                return error::NoValidTargetStatusSnafu {}.fail();
+            }
+            let custom_sigstore_metadata = json!({
+                "status": self.rekor_status,
+                "uri": self.rekor_uri,
+                "usage": "Rekor"
+            });
+
+            if let Some((target_name, target)) = rekor_target.iter_mut().next() {
+                target
+                    .custom
+                    .insert("sigstore".to_string(), custom_sigstore_metadata);
+                editor
+                    .add_target(target_name.clone(), target.clone())
+                    .context(error::DelegationStructureSnafu)?;
+            }
+        }
+        Ok(())
+    }
+
+    async fn set_tsa_target(&self, editor: &mut RepositoryEditor) -> Result<()> {
+        if let Some(ref tsa_target_path) = self.tsa_target {
+            let mut tsa_target = build_targets(tsa_target_path, self.follow).await?;
+
+            if self.tsa_status != "Active" && self.tsa_status != "Expired" {
+                return error::NoValidTargetStatusSnafu {}.fail();
+            }
+            let custom_sigstore_metadata = json!({
+                "status": self.tsa_status,
+                "uri": self.tsa_uri,
+                "usage": "Timestamp Authority"
```

Comment:
The correct usage is `"TSA"`: https://github.com/sigstore/sigstore/blob/main/pkg/tuf/usage_type.go#L37